### ISSUE TITLE
Add minor accessibility fixes to the web GUI

### ISSUE THREAD

**Issue by derekriemer** [syncthing/syncthing#3297]
I'm adding some minor accessibility fixes (mainly cosmetic) to your web gui (I'm a blind computer user and active member of the [NVDA](http://www.nvaccess.org) project) and use syncthing for maintaining many files on a subdomain of http://derekriemer.com . First off, great job on your User interface. It's really easy to understand and structured well, and because of syncthing, I have a much easier way of adding files to my website than having to jump through my many layers of authentication to connect via sftp. The minor fixes I can include are:

- There should be a heading level one somewhere on the page to comply with HTML best practices.
- Either an alt tag should be added to each of the syncthing logos that are created on line 35 of index.htm, or the span should be marked with aria-hidden="true" to convey to assistive technology to ignore these two logos.
- the language menu incorrectly has aria-expanded="true" rather than aria-expanded="false" when it starts up. Therefore, Syncthing reports "English expanded" rather than "english collapsed" as it should when the drop down is not shown.
- The actions menu does not have aria-expanded="false" on it, which causes it to report as a link, and not a link that can be expanded. This doesn't cause the link you click to activate to have any accessibility issues (The angular dropdown I think is fixing this after an expand/collapse cycle as well). Once the user activates the link, the dropdown is accessible, this is simply a cosmetic fix to alert users of Assistive Technologies that a dropdown menu exists.
- Optionally, we could add roles of menu and menuitem to each of the dropdowns code (on the ul and li tags) to turn the dropdowns into menus that appear when they are activated. If we do this, we should hide the a tags inside the list items from assistive technology.
- role="main" should be added around the area that contains the folders, and devices, to convey to assistive technology that this is the main contents.
- The folders are buttons (as of the last release), but do not have a collapsed state. I could add this by adding aria-expanded="true". I've tried this with the source, and it works great.
- We should add aria-expanded="true" to the button that displays the device name under the heading level 3. This way, assistive technology users know you can collapse this.
- Add collapsed states to each device present in the HTML code under remote devices.
- Some of the tabs on the advanced settings page need cosmetic fixes to the collapsed and expanded states. For example, currently they by default all say expanded, even if they should be collapsed. Note that after you expand and then collapse them, they take on the correct state.
- There are some modal dialogs like the simlinks are disabled that appear and don't ever get focus to set to them. Currently, a user must find them manually.

Thanks.

**Comment by derekriemer** [comment]
Does this need discussion before a pull is created?

**Comment by AudriusButkevicius** [comment]
Most of this is invisible in the visual representation, so I think we are good.

**Comment by uok** [comment]
These are all good points (except the first one IMHO is not needed, because Syncthing is not a webpage, and the different sections "folders" and "devices" are already there)

**Comment by norgeous** [comment]
related #2361

**Comment by derekriemer** [comment]
Any other a11y issues you've heard of?

Sent from a mobile device.
Please disregard errors as this is a smaller device.

**Comment by derekriemer** [comment]
Hi,
I'm possibly interested in adding a "devices" region to the devices section.

<div class="col-md-6">

I'd probably do something like

<div class="col-md-6" role="region" aria-label="Devices">

How can this be translated inside the attribute. I'm not familiar with angular-translate, so sorry for the ignorant question.
I tried:
"{{'Devices' | translate}}" and "{{Devices | translate}}" neither of which worked, I think because they point at ID's.

**Comment by norgeous** [comment]
@derekriemer not sure if you noticed, but i edited my comment above. there are three other a11y related bugs: #2305 #2361 #2697. I just searched the bug list for 'NVDA' and 'aria'.

To use the translate function on attribute use `"{{'Devices' | translate}}"` with single quotes around the string to be translated, example in `/gui/default/syncthing/usagereport/usageReportModalView.html`

It's not a fancy translator - it just swaps strings based on the language json files at `/gui/default/assets/lang/` 

I'm not sure why this is not working for you as the string 'Devices' is present in these json files.

**Comment by derekriemer** [comment]
As to http://www.oaa-accessibility.org/examplep/accordian1/ and best practices for accessibility of tab pannels, we probably should manage focus properly on this list. This is in reference to the advanced settings tab. For now, I'm implementing the easy fixes so you can merge them for the most impact, however I plan to work on this.

**Comment by derekriemer** [comment]
Actually this might be easier to fix for an accordion type widget than I thought.

**Comment by derekriemer** [comment]
Note that this work is almost done. I just have to fix one issue (haven't had Time).

**Comment by derekriemer** [comment]
https://github.com/syncthing/syncthing/pull/3463
Thanks.

**Comment by derekriemer** [comment]
Also, we should discuss setting focus in modals properly, (Some modals don't have focus put in the correct place automatically, and this might require some discussion).

**Comment by derekriemer** [comment]
Thanks.
